mpdev/internal/apply/deployment_manager.go

```diff
@@ -39,15 +37,16 @@ type DeploymentManagerAutogenTemplate struct {
 }
 
 // Apply generates a deployment manager template from an autogen file.
-func (dm *DeploymentManagerAutogenTemplate) Apply(registry *registry) error {
+func (dm *DeploymentManagerAutogenTemplate) Apply(registry Registry) error {
```

Comment:
We still want a pointer here to avoid copying. This part of go is a bit tricky. Can you check [this](https://medium.com/@saiyerram/go-interfaces-pointers-4d1d98d5c9c6) to see if my understanding is correct?

Reply:
From article:
The interface **implementation** decides the type of parameter (a pointer or a value) to pass to a method/function that accepts the interface as a parameter.

Since the struct implements the interface through pointer receiver methods, I believe all interface methods are invoked by pointer